api_server/Cargo.toml

```diff
@@ -26,7 +26,7 @@ kernel = { path = "../kernel" }
 memory_model = { path = "../memory_model" }
 net_util = { path = "../net_util" }
 rate_limiter = { path = "../rate_limiter" }
-x86_64 = { path = "../x86_64" }
+arch = { path = "../arch" }
```

Comment:
Is this a temporary state of affairs or is this the final solution for adding support for more architectures?

I am asking this because I always found the x86_64 crate to be out of place and renaming it to arch is not making the situation better. I would rather have it split in crates that have a better meaning. Memory related modules should be placed in the memory crate, bootparams should probably be in kernel (or better yet just have the HIMEM variable there since this is the only thing that we use from the generated file) and so on.

If this is just a temporary solution to kick start the arm support it is totally fine, but if we choose to go down this path with the design it would be interesting to know what is the reasoning in still grouping these under a single crate.

Reply:
@andreeaflorescu , let us brainstorm to find another name for this crate. I agree that `arch` is not the best one.